platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -213,57 +233,90 @@ passed to Event Intake as part of the event metadata.
 
 #### Topological Ordering
 
-Events are buffered if necessary to ensure that each parent event has been emitted from the Event Intake before any
-child events. A simple map (the same used for deduplication) can be used here. Given some event, for each parent, look
-up the parent by its hash. If each parent is found in the map, then emit the event. Otherwise, remember the event so
-when the missing parent is received, the child may be emitted. The current implementation uses what is known as the
-"orphan buffer" for this purpose.
+Events are buffered if necessary to ensure that each parent event has been emitted from Event Intake before any child
+events. A simple map (the same used for deduplication) can be used here. Given some event, for each parent, look up the
+parent by its hash. If each parent is found in the map, then emit the event. Otherwise, remember the event so when the
+missing parent is received, the child may be emitted. The current implementation uses what is known as the "orphan
+buffer" for this purpose.
 
 Since Event Intake will also maintain some buffers, it needs to know about the progression of the hashgraph,
 so it can evict old events. In this case, the "orphan buffer" holds events until either the parent events have
 arrived, or the events expired due to the advancement of the "non-ancient event window" and the event is
 dropped from the buffer. This document does not prescribe the existence of the orphan buffer or the method by which
 events are sorted and emitted in topological order, but it does describe a method by which old events can be dropped.
 
+#### Assigning Generations
+
+Several algorithms, including the hashgraph consensus algorithm and the tipset algorithm, require events to be part of
+a "generation". We use what is known as "local generation". Each node, when it starts, assigns "1" as the generation
+of the first event it encounters per event creator. So the first event from Alice is generation 1, as is the first event
```

Comment:
The local generation doesn't increase by 1 each time a node creates an event. Rather, the local generation is the max generation of all parents, plus 1. So if Alice builds an event with parents `a4` and `b7`, the new event will be `a8`